components/common/src/command/package/install.rs

```diff
@@ -663,14 +665,8 @@ impl<'a> InstallTask<'a> {
                    ident);
         } else if self.is_offline() {
             return Err(Error::OfflineArtifactNotFound(ident.as_ref().clone()));
-        } else if let Err(err) =
-            retry::retry_future!(delay::Fixed::from(RETRY_WAIT).take(RETRIES),
-                                 self.fetch_artifact(ui, (ident, target), token)).await
-        {
-            return Err(Error::DownloadFailed(format!("We tried {} times but \
-                                                      could not download {}. \
-                                                      Last error was: {}",
-                                                     RETRIES, ident, err)));
+        } else if let Err(err) = self.fetch_artifact(ui, (ident, target), token).await {
+            return Err(err);
         }
```

Comment:
I think this `else if let Err` could be removed and we just have `self.fetch_artifact(ui, (ident, target), token).await?`

Reply:
Added in https://github.com/habitat-sh/habitat/pull/7874/commits/2a457e8b6f09ace44d7248a5f5146dcfbd2bb44e